src/workerd/api/basics.c++

```diff
@@ -669,6 +712,97 @@ void AbortSignal::triggerAbort(
+      "AbortSignal can only be serialized for RPC.");
+  auto externalHandler = dynamic_cast<RpcSerializerExternalHander*>(&handler);
+  JSG_REQUIRE(
+      externalHandler != nullptr, DOMDataCloneError, "AbortSignal can only be serialized for RPC.");
+
+  serializer.writeRawUint32(static_cast<uint>(canceler->isCanceled()));
+  serializer.writeRawUint32(static_cast<uint>(flag));
+  KJ_IF_SOME(r, reason) {
+    serializer.write(js, r.getHandle(js));
+  } else {
+    serializer.write(js, js.undefined());
+  }
+
+  auto& ioContext = IoContext::current();
+
+  auto streamCap =
+      externalHandler->writeStream([&](rpc::JsValue::External::Builder builder) mutable {
+    auto field = builder.initAbortSignal();
+    field.setSignal(kj::heap<AbortSignalImpl>(ioContext, JSG_THIS));
+  });
+
+  // Keep track of every AbortSignal cloned from this one.
+  // If this->triggerAbort(...) is called, each rpcClient will be informed.
+  rpcClients.add(kj::mv(streamCap).castAs<rpc::AbortSignal>());
+}
+
+jsg::Ref<AbortSignal> AbortSignal::deserialize(
+    jsg::Lock& js, rpc::SerializationTag tag, jsg::Deserializer& deserializer) {
+  auto& handler = KJ_REQUIRE_NONNULL(
+      deserializer.getExternalHandler(), "got AbortSignal on non-RPC serialized object?");
+  auto externalHandler = dynamic_cast<RpcDeserializerExternalHander*>(&handler);
+  KJ_REQUIRE(externalHandler != nullptr, "got AbortSignal on non-RPC serialized object?");
+
+  auto isCanceled = static_cast<bool>(deserializer.readRawUint32());
+  auto flag = static_cast<Flag>(deserializer.readRawUint32());
+  auto reason = deserializer.readValue(js);
+
+  auto reader = externalHandler->read();
+  KJ_REQUIRE(reader.isAbortSignal(), "external table slot type does't match serialization tag");
+
+  if (isCanceled) {
+    // The signal is already aborted and cannot be triggered again. We don't need to set up RPC.
+    return abort(js, reason);
+  }
+
+  if (flag == Flag::NEVER_ABORTS) {
+    // The signal can't be aborted. We don't need to setup RPC
+    return js.alloc<AbortSignal>(/* exception */ kj::none, /* maybeReason */ kj::none, flag);
+  }
+
+  auto& ioContext = IoContext::current();
+  auto signal = js.alloc<AbortSignal>(/* exception */ kj::none, /* maybeReason */ kj::none, flag);
+
+  // The AbortSignalImpl will receive any remote triggerAbort requests and will convert this into a local call to signal->triggerAbort()
+  externalHandler->setLastStream(kj::heap<AbortSignalImpl>(ioContext, signal.addRef()));
+  return signal;
+}
+
+kj::Promise<void> AbortSignal::sendToRpc(
+    jsg::Lock& js, kj::Maybe<jsg::JsRef<jsg::JsValue>>& reason) {
+  jsg::Serializer ser(js);
+  KJ_IF_SOME(r, reason) {
+    ser.write(js, r.getHandle(js));
+  }
+
+  auto released = ser.release();
+  capnp::MallocMessageBuilder builder;
+  rpc::JsValue::Builder val = builder.initRoot<rpc::JsValue>();
+  val.setV8Serialized(kj::mv(released.data));
+  return sendToRpc(capnp::clone(val.asReader()));
```

Comment:
This `capnp::clone` seems redundant as `sendToRpc()` does not need the reader to outlive the call. You could just pass the `JsValue::Reader` itself in without cloning.

But even better yet, you could pass in the serialized data (`released.data`) instead of a whole Reader here. Then you could set it directly on each outgoing call. Then you don't need to create a `MallocMessageBuilder` at all. Two copies (and a bunch of allocation) avoided!